M2/Macaulay2/m2/code.m2

```diff
@@ -33,33 +34,34 @@ getSourceLines Sequence := x -> (
 	       ) do stop = stop + 1;
 	  if #file < stop then error("line number ",toString stop, " not found in file ", filename);
 	  while stop >= start and file#(stop-1) === "" do stop = stop-1;
-	  stack prepend(
-	       concatenate(filename, ":", 
-		    toString start, ":", toString (startcol+1),
-		    "-",
-		    toString stop, ":", toString (stopcol+1),
-		    ": --source code:"),
-	       apply(start-1 .. stop-1, i -> file#i)
-	       )
-	  )) x
+	  DIV {
+	      x, ": --source code",
+	      PRE M2CODE concatenate between_"\n" toList apply(start-1 .. stop-1, i -> file#i)
+	      }
+	  ))
 
 limit := 4
-indent := n -> "| "^(height n, depth n) | n
 
 codeFunction := (f,depth) -> (
      if depth <= limit then (
-	  if locate f === null then concatenate("function ", toString f, ": source code not available")
-	  else stack(
-	       syms := flatten \\ sortByHash \ values \ drop(localDictionaries f,-1);
-	       getSourceLines locate f,
-	       if #syms > 0 then indent listSymbols syms,
-	       if codeHelper#?(functionBody f) 
-	       then toSequence apply(
-		    codeHelper#(functionBody f) f, 
-		    (comment,val) -> indent stack (
-			      comment, 
-			      if instance(val, Function) then codeFunction(val,depth+1) else net val
-			      )))))
+	 l := locate f;
+	  if l === null then DIV{"function ", f, ": source code not available"}
```

Comment:
Is the indentation off here? Might as well also change the `if depth ...` line and fix indentation on this whole block.

Reply:
I wish we used a tool to automatically fix indentation, what I do for my other projects. will try to fix manually.

Reply:
Emacs can auto-fix indentation for M2, but I also wish there was a linter for M2.